Outlook_parser.py

```diff
@@ -0,0 +1,28 @@
+import win32com.client
+import unicodecsv as csv
+import Analise_horas
+
+output_file = open("sent_emails.csv", "wb")
+output_writer = csv.writer(output_file, delimiter=";", encoding="latin2")
+
+outlook = win32com.client.Dispatch("Outlook.Application").GetNamespace("MAPI")
+outbox = outlook.GetDefaultFolder(5)
+# "6" refers to the index of a folder - in this case,the inbox.
+# "5" refers to the index of a folder - in this case,the outbox.
+
+messages = outbox.Items
+output_writer.writerow(["Date"])
+for message in messages:
+    try:
+        # sender = message.SenderName
+        # sender_address = message.sender.address
+        # sent_to = message.To
+        date = str(message.LastModificationTime)
+        # subject = message.subject
+        print(date)
+        output_writer.writerow([date])
+    except Exception:
+        pass
```

Comment:
E' optimo que estejas a pensar em exception handling. Pelo que vejo aqui deduzo que `writerow()` possa ter uma excepcao que estas a tentar apanhar aqui. Mas nao estas a fazer nada no corpo desta excepcao, portanto nao sei qual e' o valor disto. Podias por um `print` a dizer que nao conseguiste escrever a mensagem com aquela `LastModificationTime` ou coisa parecida. Isto se nao achas que o teu programa devia crashar quando isto acontece.

Reply:
No inicio não tinha o exception handling, apesar de no exemplo de onde tirei este bocado de código tinha o exception. Como percebi que não estava a fazer nada, retirei. No entanto, quando está a processar muitos e-mails (no caso da minha caixa foram mais de 56 mil e-mails) acaba por dar erros em alguns e-mails (não sei porque nem sempre extrai a informação da caixa de correio da mesma forma) tive de voltar a colocar o except handling. O  possivel erro está na data do e-mail, nem sempre é extraido no mesmo formato. Depois no analize_data.py também tive que fazer varias adaptações ao codigo por isso mesmo, formatos diferentes neste "date". Não acho que deva crashar, porque o objetivo é ter uma ideia, uma estimativa, portanto se no meio de 56 mil emails, faltar 3 ou 4 não é grave. MAs vou adicionar um print, pelo menos fico a saber quantos mails ele "saltou".

Reply:
Ya that was my point, make it do something instead of quietly handling exceptions. E sim, isto esta a evitar que o programa crashe, duh. Nao vi isto ontem porque estava a dormir so pode.

A versao em que isto nao estaria a fazer nada would look like this:

```python
try:
    something_that_throws_exception()
except Exception
    raise Exception
```